src/openvic-simulation/misc/Define.cpp

```diff
@@ -1,109 +1,249 @@
 #include "Define.hpp"
 
-#include <cassert>
-#include <cstdlib>
-#include <memory>
-
 #include <openvic-dataloader/v2script/AbstractSyntaxTree.hpp>
 
 #include "openvic-simulation/dataloader/NodeTools.hpp"
 #include "openvic-simulation/types/Date.hpp"
 #include "openvic-simulation/types/IdentifierRegistry.hpp"
 #include "openvic-simulation/types/fixed_point/FixedPoint.hpp"
+#include "openvic-simulation/utility/StringUtils.hpp"
 
 using namespace OpenVic;
 using namespace OpenVic::NodeTools;
 
-Define::Define(std::string_view new_identifier, std::string&& new_value, Type new_type)
-	: HasIdentifier { new_identifier }, value { std::move(new_value) }, type { new_type } {}
+std::string_view Define::type_to_string(Type type) {
+	using enum Type;
+
+	switch (type) {
+	case Date: return "date";
+	case Country: return "country";
+	case Economy: return "economy";
+	case Military: return "military";
+	case Diplomacy: return "diplomacy";
+	case Pops: return "pops";
+	case Ai: return "ai";
+	case Graphics: return "graphics";
+	case Unknown: [[fallthrough]];
```

Comment:
How does this work?

Reply:
When a switch case isn't ended by a `return` or `break` the program continues through to the next case. While this is well-defined and often useful behaviour, compilers may still warn about it in case it's accidental and the author forgot to add a `break`. The [fallthrough attribute](https://en.cppreference.com/w/cpp/language/attributes/fallthrough) tells the compiler that the behaviour is intentional and suppresses any warnings.

In this case I could've just left out `case Unknown` and let it be handled by `default` directly, but IMO when using enums in switches it's better to be explicit and list every option, both to avoid ambiguity for readers and because compilers can warn about enum switch statements not handling every value (although that might only happen without a `default` statement in place).

Reply:
Funny, I thought it might indicate fallthrough to another case but then I thought "Nah, if he wanted that he would've just omitted `case Unknown:` or the `[[fallthrough]]`".

Omitted `case Unknown:` seems best here. The fallthrough attribute is best suited for implicit else statements and such if you ask me.